src/engine/strat_engine/crypt/shared.rs

```diff
@@ -838,20 +484,47 @@ pub fn activate(
             )));
         }
     }
-    log_on_failure!(
-        device.token_handle().activate_by_token::<()>(
+    if device
+        .token_handle()
+        .activate_by_token::<()>(
             Some(&name.to_string()),
-            Some(if unlock_method == UnlockMethod::Keyring {
-                LUKS2_TOKEN_ID
+            if unlock_method == UnlockMethod::Keyring {
+                Some(LUKS2_TOKEN_ID)
+            } else if unlock_method == UnlockMethod::Clevis {
+                Some(CLEVIS_LUKS_TOKEN_ID)
             } else {
-                CLEVIS_LUKS_TOKEN_ID
-            }),
+                None
+            },
             None,
             CryptActivate::empty(),
-        ),
-        "Failed to activate device with name {}",
-        name
-    );
+        )
+        .is_err()
+    {
+        if let Some(p) = passphrase {
+            let key_slot = if unlock_method == UnlockMethod::Keyring {
```

Comment:
Are you able to re-order this so that you get the keyslot number regardless of whether there is a passphrase or not, but taking into account the unlock method? The reason this seems desirable is the error messages. At present if passphrase is not specified but unlock method is wrong (the examples below assume that the pool was initially encrypted with a key in the kernel keyring and are now stopped) you get the following:

```
stratis pool start
--name spool4 --unlock-method=clevis
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: An attempt to
set up pool with UUID b935bc7e-4e40-4908-85b8-284a99b4492a from the
assembled devices failed; Device
stratis-1-private-b935bc7e4e40490885b8284a99b4492a-crypt was activated
but is reporting that it is inactive.
```

but if you also pass a passphrase:

```
stratis pool start
--name spool4 --unlock-method=clevis --capture-key
Enter passphrase followed by the return key:
Verify passphrase entered:
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: An attempt to
set up pool with UUID 158e01fd-0fed-4a94-a804-06ec263f6632 from the
assembled devices failed; Clevis keyslot not found.
```

Error messages are the same for legacy pools in the same situation.

I argue that the "Clevis keyslot not found" message is friendlier and not quite so low-level.